src/envoy/http/service_control/config_parser.cc

```diff
@@ -67,6 +67,7 @@ FilterConfigParser::FilterConfigParser(const FilterConfig& config,
   // Construct a requirement for non matched requests
   non_match_rqm_cfg_.set_service_name(first_srv_ctx->config().service_name());
   non_match_rqm_cfg_.set_operation_name(kUnrecognizedOperation);
+  non_match_rqm_cfg_.mutable_api_key()->set_allow_without_api_key(true);
```

Comment:
Why is this needed? I see the changes to `handler_impl`, but it seems like it will work without this line (?)

Reply:
Agreed, it is no need. removed.